org.eevolution.warehouse/src/main/java/base/org/eevolution/process/GenerateShipmentOutBound.java

```diff
@@ -55,216 +58,214 @@
+            });
+        }
+    }
+
+    private BigDecimal getQtyDelivered(MWMInOutBoundLine outBoundLine, BigDecimal qtyDemandDelivered) {
+        BigDecimal qtyDelivered;
+
+        if (isIncludeNotAvailable())
+            qtyDelivered = outBoundLine.getQtyToPick().subtract(qtyDemandDelivered);
+        else
+            qtyDelivered = outBoundLine.getPickedQty().subtract(qtyDemandDelivered);
+
+        return qtyDelivered;
+    }
+
+    private void processingIssues() {
+        manufacturingIssues.entrySet().stream().filter(Objects::nonNull)
+                .forEach(entry -> {
+                    MPPCostCollector issue = entry.getValue();
+                    issue.setDocAction(getDocAction());
+                    issue.processIt(getDocAction());
+                    if (!issue.processIt(getDocAction())) {
+                        addLog("@ProcessFailed@ : " + issue.getDocumentInfo());
+                        log.warning("@ProcessFailed@ :" + issue.getDocumentInfo());
+                    }
+                    issue.saveEx();
+                });
+    }
+
+    private void processingShipments() {
+        List<PO> shipmentsToPrint = new ArrayList<PO>();
+        shipments.entrySet().stream().filter(Objects::nonNull).forEach(entry -> {
+            MInOut shipment = entry.getValue();
+            shipment.setDocAction(getDocAction());
+            shipment.processIt(getDocAction());
+            if (!shipment.processIt(getDocAction())) {
+                addLog("@ProcessFailed@ : " + shipment.getDocumentInfo());
+                log.warning("@ProcessFailed@ :" + shipment.getDocumentInfo());
+            }
+            shipment.saveEx();
+            documentCreated++;
+            addLog(shipment.getDocumentNo());
+            shipmentsToPrint.add(shipment);
+        });
+        //	Print documents
+        printDocument(shipmentsToPrint, true);
+    }
+
+
+    private void processingMovements() {
+        distributionOrders.entrySet().stream().filter(entry -> entry != null).forEach(entry -> {
+            I_DD_Order distributionOrder = entry.getValue();
+            List<Integer> orderIds = new ArrayList<Integer>();
+            orderIds.add(distributionOrder.getDD_Order_ID());
+
+            ProcessInfo processInfo = ProcessBuilder.create(getCtx())
+                    .process(MovementGenerate.getProcessId())
+                    .withSelectedRecordsIds(MDDOrder.Table_ID, orderIds)
+                    .withParameter(MWMInOutBound.COLUMNNAME_M_Warehouse_ID, distributionOrder.getM_Warehouse_ID())
+                    .withParameter(MMovement.COLUMNNAME_MovementDate, getMovementDate())
+                    .withoutTransactionClose()
+                    .execute(get_TrxName());
+            if (processInfo.isError())
+                throw new AdempiereException(processInfo.getSummary());
+
+            addLog(processInfo.getSummary());
+            Arrays.stream(processInfo.getIDs()).forEach(recordId -> {
```

Comment:
Hi @e-Evolution can you use this approach instead? https://github.com/adempiere/adempiere/blob/9bd44c7573fb7bb1571fbaa11e93875af9e7d78f/org.eevolution.warehouse/src/main/java/base/org/spin/process/GenerateInvoiceInOutBound.java#L160

Is very unusefull a dialog for each document

Reply:
Also **Inventory Move Hdr (Example)** is hardcoded and not work for shipments

Reply:
I have a little change for it:

```Java
shipments.entrySet().stream().filter(entry -> entry != null).forEach(entry -> {
			MInOut shipment = entry.getValue();
			shipment.setDocStatus(MInOut.STATUS_Drafted);
			if (!shipment.processIt(getDocAction())) {
				addLog("@ProcessFailed@ : " + shipment.getDocumentInfo());
				log.warning("@ProcessFailed@ :" + shipment.getDocumentInfo());
			}
			shipment.saveEx();
			created++;
			addToMessage(shipment.getDocumentNo());
		});
		List<PO> shipmentsToPrint = new ArrayList<PO>();
		//	Print invoices
		shipments.entrySet().stream().filter(entry -> entry != null).forEach(entry -> {
			shipmentsToPrint.add(entry.getValue());
		});
		//	Print documents
		printDocument(shipmentsToPrint, true);
```